src/17D6A0.c

```diff
@@ -94,11 +98,109 @@ void free_popup(PopupMessage* popup) {
 
 INCLUDE_ASM(s32, "17D6A0", func_8024EFE0);
 
-INCLUDE_ASM(s32, "17D6A0", func_8024F394);
+void func_8024F394(void* data) {
```

Comment:
Given that this arg is interpreted as `PopupMessage*` should we type `data` as it too?
```suggestion
void func_8024F394(PopupMessage* data) {
```

Reply:
The signature needs to be this way for 2 reasons: the function is called indirectly via an argument that types the first arg of the function generically so other args can fit into it, and secondly, it doesn't match unless we create a temp for the first arg (which strongly indicates they were casting a void* to the proper type)